monads/test/Spec.hs

```diff
@@ -1,2 +1,144 @@
+import Test.Tasty
+import Test.Tasty.HUnit
+import qualified Data.Map.Strict as M
+import qualified HW.Eval
+import qualified Expr
+import qualified HW.Compiler
+import qualified HW.StackMachine
+import State.MyState (runMyState)
+
```

Comment:
There is a lot of code duplication in this module, which you can definitely trim down.